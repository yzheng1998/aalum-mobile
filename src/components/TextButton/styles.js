import styled from 'styled-components'
import theme from '../../../theme'

export const Button = styled.TouchableOpacity`
  align-self: center;
`

export const Text = styled.Text`
  color: ${theme.colors.darkGrey};
  font-family: ${theme.fonts.title};
  font-size: 16px;
  line-height: 22px;
`
