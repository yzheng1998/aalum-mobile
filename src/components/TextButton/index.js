import React, { Component } from 'react'
import { Button, Text } from './styles'

export default class TextButton extends Component {
  render() {
    const { text, ...rest } = this.props
    return (
      <Button {...rest}>
        <Text>{text}</Text>
      </Button>
    )
  }
}
