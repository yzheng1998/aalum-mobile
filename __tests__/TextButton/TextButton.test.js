import React from 'react'
import TextButton from '../../src/components/TextButton'
import { shallow } from 'enzyme'
import mountWithTheme from '../helpers/theme'

describe('TextButton', () => {
  test('renders correctly', () => {
    const wrapper = mountWithTheme(<TextButton />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ text: 'text' })
    expect(wrapper).toMatchSnapshot()
  })

  test('handles onPress correctly', () => {
    const onPressMock = jest.fn()
    const wrapper = shallow(<TextButton onPress={onPressMock} />)
    const button = wrapper.find('Button')
    button.simulate('press')

    expect(onPressMock).toHaveBeenCalledTimes(1)
  })

  test('handles disabled correctly', () => {
    const onPressMock = jest.fn()
    const wrapper = shallow(<TextButton onPress={onPressMock} disabled />)
    const button = wrapper.find('Button')

    expect(button.props().disabled).toBe(true)

    if (!button.props().disabled) {
      button.simulate('press')
    }
    expect(onPressMock).toHaveBeenCalledTimes(0)
  })
})
