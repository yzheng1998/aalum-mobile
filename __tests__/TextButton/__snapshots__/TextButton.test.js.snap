// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton renders correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <TextButton>
    <Button>
      <StyledNativeComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Button",
            "getDefaultProps": [Function],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  align-self: center;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          <AnimatedComponent
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Array [
                  Object {
                    "alignSelf": "center",
                  },
                ],
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "opacity": 1,
                  }
                }
              >
                <Styled(Text)>
                  <StyledNativeComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "displayName": "Styled(Text)",
                        "inlineStyle": InlineStyle {
                          "rules": Array [
                            "
  color: ",
                            "#8F8F8F",
                            ";
  font-family: ",
                            "Avenir-Heavy",
                            ";
  font-size: 16px;
  line-height: 22px;
",
                          ],
                        },
                        "propTypes": Object {
                          "accessible": [Function],
                          "adjustsFontSizeToFit": [Function],
                          "allowFontScaling": [Function],
                          "disabled": [Function],
                          "ellipsizeMode": [Function],
                          "minimumFontScale": [Function],
                          "nativeID": [Function],
                          "numberOfLines": [Function],
                          "onLayout": [Function],
                          "onLongPress": [Function],
                          "onPress": [Function],
                          "pressRetentionOffset": [Function],
                          "selectable": [Function],
                          "selectionColor": [Function],
                          "style": [Function],
                          "suppressHighlighting": [Function],
                          "testID": [Function],
                          "textBreakStrategy": [Function],
                        },
                        "render": [Function],
                        "styledComponentId": "StyledNativeComponent",
                        "target": [Function],
                        "withComponent": [Function],
                        Symbol(Symbol.hasInstance): [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#8F8F8F",
                            "fontFamily": "Avenir-Heavy",
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#8F8F8F",
                              "fontFamily": "Avenir-Heavy",
                              "fontSize": 16,
                              "lineHeight": 22,
                            },
                          ]
                        }
                      />
                    </Text>
                  </StyledNativeComponent>
                </Styled(Text)>
              </View>
            </View>
          </AnimatedComponent>
        </TouchableOpacity>
      </StyledNativeComponent>
    </Button>
  </TextButton>
</ThemeProvider>
`;

exports[`TextButton renders correctly 2`] = `
<ThemeProvider
  text="text"
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <TextButton>
    <Button>
      <StyledNativeComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Button",
            "getDefaultProps": [Function],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  align-self: center;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          <AnimatedComponent
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Array [
                  Object {
                    "alignSelf": "center",
                  },
                ],
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "opacity": 1,
                  }
                }
              >
                <Styled(Text)>
                  <StyledNativeComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "displayName": "Styled(Text)",
                        "inlineStyle": InlineStyle {
                          "rules": Array [
                            "
  color: ",
                            "#8F8F8F",
                            ";
  font-family: ",
                            "Avenir-Heavy",
                            ";
  font-size: 16px;
  line-height: 22px;
",
                          ],
                        },
                        "propTypes": Object {
                          "accessible": [Function],
                          "adjustsFontSizeToFit": [Function],
                          "allowFontScaling": [Function],
                          "disabled": [Function],
                          "ellipsizeMode": [Function],
                          "minimumFontScale": [Function],
                          "nativeID": [Function],
                          "numberOfLines": [Function],
                          "onLayout": [Function],
                          "onLongPress": [Function],
                          "onPress": [Function],
                          "pressRetentionOffset": [Function],
                          "selectable": [Function],
                          "selectionColor": [Function],
                          "style": [Function],
                          "suppressHighlighting": [Function],
                          "testID": [Function],
                          "textBreakStrategy": [Function],
                        },
                        "render": [Function],
                        "styledComponentId": "StyledNativeComponent",
                        "target": [Function],
                        "withComponent": [Function],
                        Symbol(Symbol.hasInstance): [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#8F8F8F",
                            "fontFamily": "Avenir-Heavy",
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#8F8F8F",
                              "fontFamily": "Avenir-Heavy",
                              "fontSize": 16,
                              "lineHeight": 22,
                            },
                          ]
                        }
                      />
                    </Text>
                  </StyledNativeComponent>
                </Styled(Text)>
              </View>
            </View>
          </AnimatedComponent>
        </TouchableOpacity>
      </StyledNativeComponent>
    </Button>
  </TextButton>
</ThemeProvider>
`;
